### PR TITLE
Support assignment to FieldExpr for extract_assignment assist

### PR DESCRIPTION
**Pull request by Jesse-Bakker**

### CODE REVIEW

File: crates/assists/src/handlers/extract_assignment.rs [hunk 1]
```diff
@@ -117,6 +120,23 @@ fn exprify_block(block: &ast::BlockExpr, name: &hir::Name) -> Option<ast::BlockE
     None
 }
 
+fn eq_name_expr(expr0: &ast::Expr, expr1: &ast::Expr) -> Option<bool> {
+    match (expr0, expr1) {
+        (ast::Expr::FieldExpr(field_expr0), ast::Expr::FieldExpr(field_expr1)) => {
+            let path0 = ast::PathExpr::cast(field_expr0.expr()?.syntax().to_owned())?;
```
Comment (matklad) [inline crates/assists/src/handlers/extract_assignment.rs:126]:
```suggestion
            mark::hit!(test_extract_assignment_field_assignment);
            let path0 = ast::PathExpr::cast(field_expr0.expr()?.syntax().to_owned())?;
```

**Comment by Jesse-Bakker** [comment]
@matklad I was thinking that it may be nicer to compare the paths with `path.segment()?.name_ref()?.as_name()`, but I'm not sure if that does what I want.

**Comment by matklad** [comment]
@Jesse-Bakker I think that the right solution might be comparing `sema.resolve_field`s perhaps? But this will also have edge cases....

I think we need to extract

```rust
fn is_equivalent(sema: &Semantics, e1: &ast::Expr, e2: &ast::Expr) -> bool
```

and then tweak the impl as we go forward

**Pull request merged**